test/spec/functions/defaultsPure.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+var clone = require('lodash/clone');
+var defaults = require('../../../src/functions/defaultsPure');
+
+// tests modified from lodash source
+
+it('should assign source properties if missing on `object`', function() {
+  var actual = defaults({'a': 1}, {'a': 2, 'b': 2});
+  expect(actual).toEqual({'a': 1, 'b': 2});
+});
+
+it('should accept multiple sources', function() {
+  var expected = {'a': 1, 'b': 2, 'c': 3};
+  var actual = defaults({'a': 1, 'b': 2}, {'b': 3}, {'c': 3});
+
+  expect(actual).toEqual(expected);
+
+  actual = defaults({'a': 1, 'b': 2}, {'b': 3, 'c': 3}, {'c': 2});
+  expect(actual).toEqual(expected);
+});
+
+it('should not overwrite `null` values', function() {
+  var actual = defaults({'a': null}, {'a': 1});
+  expect(actual.a).toBe(null);
+});
+
+it('should overwrite `undefined` values', function() {
+  var actual = defaults({'a': undefined}, {'a': 1});
+  expect(actual.a).toBe(1);
+});
+
+it('should assign `undefined` values', function() {
+  var source = {'a': undefined, 'b': 1};
+  var actual = defaults({}, source);
+
+  expect(actual).toEqual({'a': undefined, 'b': 1});
```

Comment:
Shouldn't this be like this?
```js
expect(actual).toEqual({'b': 1});
```

Reply:
this test was copied from lodash, but explicitly putting undefined on an object still matches toEqual I think

Reply:
Ah okay. I just tested and this passes.

```js
expect({ b: 1 }).toEqual({ a: undefined, b: 1 });
```